scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -95,11 +97,54 @@ namespace <%= namespace %> {
         }
 
         public void CheckoutWithNativeWebView(CheckoutSuccessCallback success, CheckoutFailureCallback failure, CheckoutCancelCallback cancelled) {
-            GetWebCheckoutLink(url => {
-                WebCheckout.Checkout(url, success, failure, cancelled);
-            }, error => {
-                failure(error);
-            });
+            if (WebCheckout != null) {
+                GetWebCheckoutLink(url => {
+                    WebCheckout.Checkout(url, success, failure, cancelled);
+                }, error => {
+                    failure(error);
+                });
+            } else {
+                // TODO throw an exception,  this platform does not support checkout with web pay
+            }
+        }
+
+        public void CheckoutWithNativePay(string key, CheckoutSuccessCallback success, CheckoutFailureCallback failure, CheckoutCancelCallback cancelled) {
+            if (NativeCheckout != null) {
+                CheckoutSave(error => {
+                    if (error != null) {
+                        failure(error);
+                    }
+                    else {
+                        NativeCheckout.Checkout(key, success, failure, cancelled);
+                    }
+                });
+            } else {
+                // TODO throw an exception, this platform does not support checkout with native pay
+            }
+        }
+
+        public bool CanShowNativePaySetup() {
```

Comment:
NativePay Setup shows the iOS `Wallet` app, and instructs the user to add a card to Apple Pay. You won't be able to checkout if the user has no cards added to `Apple Pay`